### PR TITLE
test: event-native integration test cases

### PR DESCRIPTION
**Pull request by ytvnr**
- sse entrypoint with kafka endpoint
- http-post with kafka endpoint
- http-get with kafka endpoint

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-tgzpzvgcok.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/en-message-usecases/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-tests-integration/src/test/java/io/gravitee/gateway/tests/sme/HttpPostEntrypointKafkaEndpointIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    @Override
+    public void configureEntrypoints(Map<String, EntrypointConnectorPlugin<?, ?>> entrypoints) {
+        entrypoints.putIfAbsent("http-post", EntrypointBuilder.build("http-post", HttpPostEntrypointConnectorFactory.class));
+    }
+
+    @Override
+    protected void configureGateway(GatewayConfigurationBuilder gatewayConfigurationBuilder) {
+        gatewayConfigurationBuilder.set("api.jupiterMode.enabled", "true");
+        gatewayConfigurationBuilder.set("api.jupiterMode.default", "always");
+    }
```
Comment (ytvnr) [inline gravitee-apim-gateway/gravitee-apim-gateway-tests-integration/src/test/java/io/gravitee/gateway/tests/sme/HttpPostEntrypointKafkaEndpointIntegrationTest.java]: to remove, defautl behavior with apim 4.0.0-SNAPSHOT

File: gravitee-apim-gateway/gravitee-apim-gateway-tests-integration/src/test/java/io/gravitee/gateway/tests/messages/SseEntrypointKafkaEndpointIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            })
+            .test();
+
+        // We expect 4 chunks, 1 retry message and 3 messages
+        obs
+            .awaitCount(4)
+            .assertValueAt(
+                0,
+                chunk -> {
+                    System.out.println(chunk);
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-tests-integration/src/test/java/io/gravitee/gateway/tests/messages/SseEntrypointKafkaEndpointIntegrationTest.java]: I recognize a residue of reactive debugging session 😜

Comment (ytvnr) [reply gravitee-apim-gateway/gravitee-apim-gateway-tests-integration/src/test/java/io/gravitee/gateway/tests/messages/SseEntrypointKafkaEndpointIntegrationTest.java]: I made some console.log() in a js debugging session earlier 😄 , same reflex

File: gravitee-apim-plugin/gravitee-apim-plugin-tests-integration/src/test/java/io/gravitee/plugin/tests/messages/HttpGetEntrypointKafkaEndpointIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    void shouldBeAbleToSubscribeToKafkaEndpointWithHTTPGetEntrypoint(HttpClient client, Vertx vertx) {
+        // In order to simplify the test, Kafka endpoint's consumer is configured with "autoOffsetReset": "earliest"
+        // It allows us to publish the messages in the topic before opening the api connection through SSE entrypoint.
+        KafkaProducer<String, byte[]> producer = getKafkaProducer(vertx);
+        final RecordMetadata recordMetadata1 = publishMessage(producer, "message1");
+        final RecordMetadata recordMetadata2 = publishMessage(producer, "message2");
+        final RecordMetadata recordMetadata3 = publishMessage(producer, "message3");
+        producer.close();
+
+        client
```
Comment (leleueri) [inline gravitee-apim-plugin/gravitee-apim-plugin-tests-integration/src/test/java/io/gravitee/plugin/tests/messages/HttpGetEntrypointKafkaEndpointIntegrationTest.java:123]: remark: another call to get the third message could be nice. wdyt?

Comment (ytvnr) [reply gravitee-apim-plugin/gravitee-apim-plugin-tests-integration/src/test/java/io/gravitee/plugin/tests/messages/HttpGetEntrypointKafkaEndpointIntegrationTest.java:123]: as discussed, this case is not doable now :(

**Review by leleueri — APPROVED** [review]

**Pull request merged**